### DISCUSSION TITLE
interpretation of units in cocip.contrial output table?

### DISCUSSION THREAD

**Discussion by EmilieSimone** [contrailcirrus/pycontrails#116] [Q&A] [answered]
I am trying to use cocip to calculate an average global radiative forcing per flown km for a flight.

However i am unsure of where i can find the length of a flight route and how to interpret the RF_net column?

I went through the following tutporial: [2023-04-pycontrails-workshop](https://github.com/contrailcirrus/2023-04-pycontrails-workshop/tree/main)/[notebooks](https://github.com/contrailcirrus/2023-04-pycontrails-workshop/tree/main/notebooks)
/03-CoCiP-single.ipynb

I am interested in the output from cell 19, which gives the following table. Here my idea is to use the RF_net column to find an average rf for the flight in terms of RF_net/km flown.

![image](https://github.com/contrailcirrus/pycontrails/assets/152161890/80a06404-e2f9-4012-ad4a-fe98faf644ed)

Further I could not find any description of the units for this specific table output. Do anyone know where I can find it? Or is it the same as in Schumann 2012 article, where the model is presented?

**Comment by zebengberg** [comment, marked as answer]
I've tried to better document the `cocip.contrail` DataFrame columns you're investigating: https://py.contrails.org/api/pycontrails.models.cocip.Cocip.html#pycontrails.models.cocip.Cocip.contrail
I didn't get to every single column.

@roger-teoh , can you suggest the best way to compute RF_net / km_flown for a single flight?

**Reply by EmilieSimone** [reply]
Thank you very much @zebengberg, that is just what I was looking for!

I found this piece of code i tutorial 02-flights, which should be sufficient.
I assume it is the actual amount of flown meters i cruise altitude, and not the equivilent distance at surface level. 
Do you know if this is a correct assumption?

![image](https://github.com/contrailcirrus/pycontrails/assets/152161890/e48f7633-040b-4437-8439-e88d4eb50719)

**Reply by zebengberg** [reply]
It **is** the distance at surface level. This distinction doesn't vastly change the output:

```python
>>> from pycontrails.physics import constants, geo

>>> print(geo.haversine(0, 0, 50, 50))
7294149.279093339

>>> constants.radius_earth += 3000  # increase radius of earth by 3 km
>>> print(geo.haversine(0, 0, 50, 50))
7297583.851581203
```
